### PR TITLE
remove test-encoding job

### PR DESCRIPTION
**Pull request by frenchy64**
We can remove a lot of code and save electricity by booting up a new JVM just for the encoding test.

Takes 43s instead of 2m if we let the encoding tests be balanced with the rest of the tests. We can reuse docker and probably don't need to compile as much code. The test matrix seems to take the same amount of time.

https://github.com/threatgrid/ctia/actions/runs/11637562591/job/32411277250?pr=1453#step:8:308

<a name="qa">[§](#qa)</a> QA
============================

No QA is needed.

<a name="release-notes">[§](#release-notes)</a> Release Notes
=============================================================

```
intern: remove test-encoding job
```

<a name="squashed-commits">[§](#squashed-commits)</a> Squashed Commits
======================================================================

### CODE REVIEW

File: test/ctia/test_helpers/core.clj [hunk 1]
```diff
@@ -68,14 +69,13 @@
...
          "ctia.versions.config"                       "test"]
-        ;; use es7 if es5 is not available
         (let [es-versions (set-of-es-versions-to-test)]
-          (when (es-versions 7)
-            ["ctia.store.es.default.port" 9207
-             "ctia.store.es.default.auth.type" "basic-auth"
-             "ctia.store.es.default.auth.params.user" "elastic"
-             "ctia.store.es.default.auth.params.pwd" "ductile"
-             "ctia.store.es.default.version" 7]))))
+          (assert (es-versions 7))
```
Comment (ereteog) [inline test/ctia/test_helpers/core.clj:73]: we may add ES8 this year.

**Review by ereteog — APPROVED** [review]
LGTM

File: test/ctia/test_helpers/core.clj (hunk 1 above)
Comment (frenchy64) [reply test/ctia/test_helpers/core.clj:73]: Ah, I will revise this PR to allow for that.

**Review by DeLaGuardo — APPROVED** [review]

File: .github/workflows/pr.yml [hunk 2]
```diff
@@ -59,7 +60,7 @@ jobs:
           mkdir -p "${BIN_PATH}"
           echo "${BIN_PATH}" >> $GITHUB_PATH
       - name: Install clojure tools
-        uses: DeLaGuardo/setup-clojure@12.5
+        uses: DeLaGuardo/setup-clojure@2d31e3ee6005571f8b20ab9c0827f29238f58d88
```
[the remark this answers is not in the document]
Comment (frenchy64) [reply .github/workflows/pr.yml:63]: Thanks, it's blazing fast!

**Comment by frenchy64** [comment]
I'm not happy with the added complexity, I will think about another approach if any.

**Pull request closed without merging**